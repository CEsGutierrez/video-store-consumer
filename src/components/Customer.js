import React from 'react';
import PropTypes from 'prop-types';

const Customer = ({id, name, registered_at, address, city, state, postal_code, phone, account_credit, movies_checked_out_count, selectCustomerCallback}) => {
  return (
    <div>
      {name}
      <button
          // className="btn btn-primary pet-card--select-pet-btn"
          onClick={() => { selectCustomerCallback(id, 'customerList', 'currentCustomer' ) }}
        >
          Select
      </button>
    </div>
  )
}

Customer.propTypes = {
  id: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
  registered_at: PropTypes.string.isRequired,
  address: PropTypes.string.isRequired,
  city: PropTypes.string.isRequired,
  state: PropTypes.string.isRequired,
  postal_code: PropTypes.string.isRequired,
  phone: PropTypes.string.isRequired,
  account_credit: PropTypes.number.isRequired,
  movies_checked_out_count: PropTypes.number.isRequired,
  selectCustomerCallback: PropTypes.func.isRequired,
};

export default Customer;