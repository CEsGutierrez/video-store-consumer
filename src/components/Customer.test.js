import React from 'react';
import { render, cleanup } from '@testing-library/react'
import Customer from './Customer';

describe('Customer', () => {
  test('that it matches the existing snapshot', () => {
    // Arrange-Act
    const { asFragment } = render(
      <Customer
      id={1}
      name=''
      registered_at=''
      address=''
      city=''
      state=''
      postal_code=''
      phone=''
      account_credit={3.0}
      movies_checked_out_count={3}
      selectCustomerCallback={() => { }}
      />
    );

    // Assert
    expect(asFragment()).toMatchSnapshot();
    cleanup();
  });

  test('It will render the proper customer fields', () => {
    // Arrange & Act
    const container = render(<Customer
      id={1}
      name='Nicolas Cage'
      registered_at=''
      address='Avatar'
      city=''
      state=''
      postal_code=''
      phone=''
      account_credit={3.0}
      movies_checked_out_count={3}
      selectCustomerCallback={() => { }}
      />);

    // Assert
    expect(container.getByText(/Nicolas Cage/)).toBeDefined();
  });

  test('The "selectCustomerCallback" function is called when the `select` button is clicked on', () => {

    const selectCustomer = jest.fn();

    const container = render(<Customer
      id={1}
      name=''
      registered_at=''
      address=''
      city=''
      state=''
      postal_code=''
      phone=''
      account_credit={3.0}
      movies_checked_out_count={3}
      selectCustomerCallback={selectCustomer}
      />);

    const selectButton = container.getByText(/Select/);
    selectButton.click();

    expect(selectCustomer).toHaveBeenCalled();
  });

});