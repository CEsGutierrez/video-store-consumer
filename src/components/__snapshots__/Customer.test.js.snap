// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Customer that it matches the existing snapshot 1`] = `
<DocumentFragment>
  <div>
    <button>
      Select
    </button>
  </div>
</DocumentFragment>
`;
